fixed inconsistent output issue

sumanbyte opened the pull request:
### Description

In addressing a problem, I discovered that:

new BikramSambat()
BikramSambat.fromAD(new Date())
were producing inconsistent results.

The issue arose because, when creating a new instance of BikramSambat, the current date was converted to an ISO string and then sliced to reset the UTC hours to 00:00:00. However, when using the static method directly (BikramSambat.fromAD(new Date())), the UTC handling led to slight discrepancies, causing the date calculation to be offset by one day.

### Issue Reference

Inconsistent output https://github.com/askbuddie/bikram-sambat/issues/48

### Proposed Changes

To ensure consistent results between both approaches, I've applied uniform UTC handling for date calculations, maintaining the same logic for resetting the date to 00:00:00 in UTC.

### Screenshots / GIFs / Videos


### Checklist

- [x] Lint rules are passed correctly
- [x] Code builds correctly
- [x] Created or updated tests for the new code
- [x] All existing and new tests passed
- [x] Pull request title is concise and descriptive.
- [x] Pull request is linked to the related issue.

ashiishme reviewed (commented):
@sumanbyte  Thank you so much for taking care of this issue. It's been a long time since I was involved in this project, I totally forgot why we slice the `TimeString` in the constructor.

I think rather than resetting the `TimeString` in this PR, I guess it's fine to just remove the slicing from the constructor.

You can do that in this PR and try to run the test and see if all the existing test passes. If yes, you can then add a new test to check if both dates are equal as stated by the original problem.

Let me know what do you think?

sumanbyte commented:
As suggested. i removed the slicing operation from the constructor. instead of using the setUTCHours for consistent output incase of partial days. i changed the getDaysBetweenTwoAdDates function to ignore partial days by using math.floor function instead of math.ceil to correctly count the number of days. i wrote test cases for partial days by checking if it outputs 0 incase of the same current date but in the end date adding 5 minutes to it. Another test case i added was to check if both instance date string and static method date string both matches or not.

ashiishme commented on src/BikramSambat.ts:
```diff
@@ -181,7 +181,13 @@ export class BikramSambat {
+
     const gregorianDate = new Date(date)
+
+    if (isNaN(gregorianDate.getTime())) {
+      return new BikramSambat(InvalidDate)
+    }
+
     if (gregorianDate.toString() === InvalidDate) {
       return new BikramSambat(InvalidDate)
     }
```
Do we need this check `isNaN(gregorianDate.getTime())` btw, you can merge these two condition with OR operator.

sumanbyte replied:
it is an extra safety i think. so I've put there. i'll merge with OR.

ashiishme commented on src/index.ts:
```diff
@@ -1,3 +1,4 @@
 
-export default BikramSambat
+
+export default BikramSambat
```
I think you can revert this file, doesn't include any changes.

sumanbyte replied:
Done.

ashiishme commented on src/utils/getDaysBetweenTwoAdDates.test.ts:
```diff
@@ -18,4 +18,11 @@ describe('getDaysBetweenTwoAdDates', () => {
     expect(getDaysBetweenTwoAdDates(startDate, endDate)).toBe(0)
   })
+
+  it('calculates the correct number of days ignoring partial days', () => {
```
Do you have an example why we need this?

sumanbyte replied:
Suppose if there is a new Date() instance created and for some reason we passed a date having some minutes added to it. in that case our code should output number of days ignoring that minute value considering it as full day. Since i've also changed the function to prevent this case. I thought it would be a good idea to put this in a test case so that consistency remains there.

ashiishme replied:
Ah I see. Thank you so much for the refactoring & the test.

ashiishme commented:
Thank you again @sumanbyte for taking care of this issue.
